allennlp/training/metrics/categorical_accuracy.py

```diff
@@ -44,18 +52,29 @@ def __call__(self,
             raise ConfigurationError("A gold label passed to Categorical Accuracy contains an id >= {}, "
                                      "the number of classes.".format(num_classes))
 
-        # Top K indexes of the predictions (or fewer, if there aren't K of them).
-        # Special case topk == 1, because it's common and .max() is much faster than .topk().
-        if self._top_k == 1:
-            top_k = predictions.max(-1)[1].unsqueeze(-1)
-        else:
-            top_k = predictions.topk(min(self._top_k, predictions.shape[-1]), -1)[1]
+        predictions = predictions.view((-1, num_classes))
+        gold_labels = gold_labels.view(-1).long()
+        if not self._tie_break:
+            # Top K indexes of the predictions (or fewer, if there aren't K of them).
+            # Special case topk == 1, because it's common and .max() is much faster than .topk().
+            if self._top_k == 1:
+                top_k = predictions.max(-1)[1].unsqueeze(-1)
+            else:
+                top_k = predictions.topk(min(self._top_k, predictions.shape[-1]), -1)[1]
 
-        # This is of shape (batch_size, ..., top_k).
-        correct = top_k.eq(gold_labels.long().unsqueeze(-1)).float()
+            # This is of shape (batch_size, ..., top_k).
+            correct = top_k.eq(gold_labels.unsqueeze(-1)).float()
+        else:
+            # prediction is correct if gold label falls on any of the max scores. distribute score by tie_counts
+            max_predictions = predictions.max(-1)[0]
+            max_predictions_mask = predictions.eq(max_predictions.unsqueeze(-1).expand_as(predictions))
+            correct = max_predictions_mask[torch.arange(gold_labels.numel()).long(), gold_labels].float()
+            tie_counts = max_predictions_mask.sum(-1)
+            correct /= tie_counts.float()
+            correct.unsqueeze_(-1)
```

Comment:
If you remove this unsqueeze, you could replace the view below to be `.view(-1)` or something.

Reply:
Yeah, but I didn't do that because of the following line above it (in **if clause**)
```
            # This is of shape (batch_size, ..., top_k).
            correct = top_k.eq(gold_labels.unsqueeze(-1)).float()
```
It is necessary to have last dimension unsqueezed here because `top_k` might be `batch_size X 1` or `batch_size X top_k`. Due to this I keep `correct` in **else clause** also unsqueezed so that I don't need to handle these things separately in `correct *= mask.view(-1, 1).float()`. Overall this ends up being cleaner.

Reply:
Sure, sounds good.